src/Twig/WidgetExtension.php

```diff
@@ -13,23 +13,40 @@
  */
 class WidgetExtension extends AbstractExtension
 {
+    /** @var Snippets */
+    private $snippets;
+
+    public function __construct(Snippets $snippets)
+    {
+        $this->snippets = $snippets;
+    }
     /**
      * {@inheritdoc}
      */
-    public function getFunctions()
+    public function getFunctions(): array
     {
         $safe = ['is_safe' => ['html']];
-        $env = ['needs_environment' => true];
 
         return [
-            new TwigFunction('countwidgets', [$this, 'dummy'], $safe + $env),
-            new TwigFunction('getwidgets', [$this, 'dummy'], $safe),
-            new TwigFunction('haswidgets', [$this, 'dummy'], $safe + $env),
-            new TwigFunction('widgets', [$this, 'dummy'], $safe + $env),
+            new TwigFunction('countwidgets', [$this, 'dummy'], $safe),
+            new TwigFunction('listwidgets', [$this, 'dummy'], $safe),
+            new TwigFunction('haswidgets', [$this, 'dummy'], $safe),
+            new TwigFunction('widgets', [$this, 'getWidgets'], $safe),
```

Comment:
rename to render_widgets

Reply:
Not a fan of that.. `{{ widgets('main_top') }}` is much nicer than `{{ render_widgets('main_top') }}`

Reply:
`widgets` says nothing about what it does...

Reply:
I did change the method though:

```php
new TwigFunction('widgets', [$this, 'renderWidgets'], $safe),

…

    public function renderWidgets(string $target): string
    {
        return $this->snippets->renderWidgets($target);
    }
```

Reply:
Looks like you forgot to push your changes ;)